### PR TITLE
[PPP-4182] Use of vulnerable component dom4j-1.6.1.jar  CVE-2018-1000632

### PR DESCRIPTION
**Pull request by pentaho-lmartins**
This is one of a series of PRs. See pentaho/maven-parent-poms#70 for the complete list and more details. **Do not merge until further notice!**

@graimundo

### CODE REVIEW

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java [hunk 1]
```diff
@@ -89,6 +89,8 @@
 
 import com.thoughtworks.xstream.XStream;
 
+import static java.util.stream.Collectors.toList;
```
Comment (mbatchelor) [inline core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java]: Static imports are anti-style for non-test cases.

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java [hunk 2]
```diff
@@ -579,13 +581,8 @@ public DatasourceDTO deSerializeModelState( String dtoStr ) throws DatasourceSer
...
       component.setAction( PMDUIComponent.ACTION_LIST_MODELS );
       Document document = component.getXmlContent();
-      List<DefaultElement> modelElements = document.selectNodes( "//model_name" ); //$NON-NLS-1$
 
-      ArrayList<String> datasourceNames = new ArrayList<String>();
-      for ( DefaultElement element : modelElements ) {
-        datasourceNames.add( element.getText() );
-      }
-      return datasourceNames;
+      return ( (List<Node>) document.selectNodes( "//model_name" ) ).stream().map( x -> x.getText() ).collect( toList() );
```
Comment (mbatchelor) [inline core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java]: While correct, this diminishes readability. Not sure this was warranted.

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java (hunk 1 above)
Comment (pentaho-lmartins) [reply core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java]: See: https://github.com/pentaho/data-access/pull/1021

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java (hunk 2 above)
Comment (pentaho-lmartins) [reply core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java]: See: https://github.com/pentaho/data-access/pull/1021

**Pull request merged**